### PR TITLE
[sgd] Tune interface for Pytorch MultiNode SGD

### PR DESCRIPTION
**Pull request by jichan3751**
## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by jichan3751** [comment]
1. In test_train() which was originally at place, it checks whether train loss and validation loss is decreasing. In order to do this, I'm checking losses from analysis.trial_dataframes. Let me know if this is not the right way to be done.

2. Also, let me know if any other tune test to be done in test_pytorch.py

3. At examples/train_example.py, after trials are done, analysis.get_best_config() is not working properly since it only considers 1 trial and other 11 trials are not considered, because it uses _get_trial_paths() method override at class ExperimentAnalysis in ray.tune.experiment_analysis. Looks like this method only considers single trial from multiple trials. Can you explain me the purpose of this method and way to get around this?

4. NCCL bug
Not yet fixed. I tried pytorch version 1.2.0 which reflects [](https://github.com/pytorch/pytorch/pull/13496,) but it still hangs at the same point that we've seen last time.

**Comment by richardliaw** [comment]
Ah you should set `tune.run(global_checkpoint_period=0)`.

File: python/ray/experimental/sgd/pytorch/pytorch_trainer.py [hunk 1]
```diff
@@ -152,3 +166,43 @@ def shutdown(self):
...
+            backend=config["backend"])
+
+    def _train(self):
+
+        train_stats = self._trainer.train()
+        validation_stats = self._trainer.validate()
+
+        train_stats.update(validation_stats)
+
+        # output {"mean_loss": test_loss, "mean_accuracy": accuracy}
```
Comment (richardliaw) [inline python/ray/experimental/sgd/pytorch/pytorch_trainer.py]: can you clean this up?

**Comment by jichan3751** [comment]
> Ah you should set `tune.run(global_checkpoint_period=0)`.

We need to add this, right?

**Review by richardliaw — APPROVED** [review]
